src/prompt/engine.go

```diff
@@ -510,32 +512,38 @@ func (e *Engine) adjustTrailingDiamondColorOverrides() {
 	}
 }
 
-func (e *Engine) checkPromptCache() bool {
-	data, ok := e.Env.Cache().Get(cache.PROMPTCACHE)
+func (e *Engine) restoreEngineFromCache() bool {
+	if !e.Env.Flags().Cached {
```

Comment:
This flag is for the primary prompt only. Once a tooltip is rendered, we always cache it for reuse, regardless of this flag. Since we don't set the flag as `true` when requesting a tooltip from OMP CLI, if you check it here, a tooltip will never be cached.

Reply:
But should we do that for PowerShell then?

Reply:
For PowerShell only? Yes, at least currently this approach of prompt caching is used in PowerShell only.

Reply:
I would let the CLI determine this rather than a check in the code. That way it's reusable whenever.